Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue

```diff
@@ -0,0 +1,457 @@
+);
+const isDependentAlreadyAdded = computed(() =>
+    dependentStore.dependents.some(
+        (d) =>
+            d.dependentInformation.PHN ===
+            dependent.value.PHN.replace(/\s/g, "")
+    )
+);
+const minBirthdate = computed<IDateWrapper>(() =>
+    new DateWrapper().subtract(
+        Duration.fromObject({ years: configStore.webConfig.maxDependentAge })
+    )
+);
+const validations = computed(() => ({
+    dependent: {
+        firstName: {
+            required: helpers.withMessage(
+                "Given names are required.",
+                required
+            ),
+        },
+        lastName: {
+            required: helpers.withMessage("Last name is required.", required),
+        },
+        dateOfBirth: {
+            required: helpers.withMessage("Invalid date.", required),
+            minLength: helpers.withMessage("Invalid date.", minLength(10)),
+            minValue: helpers.withMessage(
+                `Dependent must be under the age of
+                        ${configStore.webConfig.maxDependentAge}.`,
+                (value: string) =>
+                    new DateWrapper(value).isAfter(minBirthdate.value)
+            ),
+            maxValue: helpers.withMessage("Invalid Date.", (value: string) =>
+                new DateWrapper(value).isBefore(new DateWrapper())
+            ),
+        },
+        PHN: {
+            required: helpers.withMessage("Valid PHN is required.", required),
+            minLength: helpers.withMessage(
+                "Valid PHN is required.",
+                minLength(12)
+            ),
+            validPersonalHealthNumber: helpers.withMessage(
+                "Personal Health Number must be valid.",
+                ValidationUtil.validatePhn
+            ),
+            isNew: helpers.withMessage(
+                "This dependent has already been added.",
+                () => !isDependentAlreadyAdded.value
+            ),
+        },
+    },
+    accepted: { isChecked: sameAs(true) },
+}));
+
+const v$ = useVuelidate(validations, { dependent, accepted });
+
+function addDependent(): void {
+    isLoading.value = true;
+    dependentService
+        .addDependent(userStore.hdid, {
+            ...dependent.value,
+            PHN: dependent.value.PHN.replace(/\D/g, ""),
+        })
+        .then(async () => {
+            errorType.value = null;
+
+            await nextTick();
+
+            hideDialog();
+            emit("handle-submit");
+        })
+        .catch((err: ResultError) => {
+            if (err.statusCode === 429) {
+                setTooManyRequestsError("addDependentDialog");
+            } else {
+                errorMessage.value = err.resultMessage;
+                errorType.value = err.actionCode ?? null;
+            }
+        })
+        .finally(() => (isLoading.value = false));
+}
+
+function clear(): void {
+    dependent.value = {
+        firstName: "",
+        lastName: "",
+        dateOfBirth: "",
+        PHN: "",
+    };
```

Comment:
The empty dependent value is used twice, this can be moved to its on variable for reuse

```
const emptyDependent = { 
    //... your model here 
}

dependent.value = {...emptyDependent}
```

Reply:
Changed.